### PR TITLE
Component isolation for donation widget

### PR DESCRIPTION
**Pull request by duranmla**
**What:**
Separate the code of the forms that we are going to have to accomplish the donation widget

**Why:**
The idea is to reduce the noise when the aesthetics PR come

**How:**
- Create dummy components to hold form compositions for each step of the widget
- Fix the `UPDATE.AMOUNT` event in order to have one for `MONTHLY` and `ONCE` donation type

**Media:**
Doesn't represent any visual change only tech refactor preparing the code.

### CODE REVIEW

File: src/components/DonationWidget/components/DonationAddressForm.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+
+const DonationAddressForm = ({
```
Comment (arathjz) [inline src/components/DonationWidget/components/DonationAddressForm.tsx:3]: 💯

Comment (arathjz) [inline src/components/DonationWidget/components/DonationAddressForm.tsx:3]: In order to keep the same pattern from other components, I'd suggest declaring prop types as following

```suggestion
interface Props {
  defaultValues: {
    address: string;
    city: string;
    zipCode: string;
    country: string;
  };
  onSubmit: (e: React.ChangeEvent<HTMLFormElement>) => void;
};

const DonationAddressForm: React.FC<Props> = ({defaultValues,
  onSubmit}) => {
 ....
}
```

**Review by arathjz — COMMENTED** [review]
LGMT, the only change I'd suggest is to declare props in the same way than other components to keep consistency.

We could also use your approach, I just think it's pretty important for the components to keep the same pattern

File: src/components/DonationWidget/components/DonationAddressForm.tsx (hunk 1 above)
Comment (duranmla) [reply src/components/DonationWidget/components/DonationAddressForm.tsx:3]: I will say you are totally on point in regards keeping the consistency. _I will argue to use types instead interface, but if this is the approach we have lets proceed with it and then we can challenge if necessary_
_I didn't know about this feature on Github_

Comment (duranmla) [reply src/components/DonationWidget/components/DonationAddressForm.tsx:3]: I won't commit from here to make a commit to address this on all components I have introduced.

**Pull request merged**